api/v1alpha3/gcpmachine_types.go

```diff
@@ -33,8 +33,25 @@ type DiskType string
 const (
 	PdStandardDiskType DiskType = "pd-standard"
 	PdSsdDiskType      DiskType = "pd-ssd"
+	LocalSsdDiskType   DiskType = "local-ssd"
 )
 
+// AttachedDiskSpec degined GCP machine disk.
+type AttachedDiskSpec struct {
+	// DeviceType is a device type of the attached disk.
+	// Supported types of non-root attached volumes:
+	// 1. "pd-standard" - Standard (HDD) persistent disk
+	// 2. "pd-ssd" - SSD persistent disk
+	// 3. "local-ssd" - Local SSD disk (https://cloud.google.com/compute/docs/disks/local-ssd).
+	// Default is "pd-standard".
+	// +optional
+	DeviceType *DiskType `json:"deviceType,omitempty"`
+	// Size is the size of the disk in GBs.
+	// Defaults to 30GB. For "local-ssd" size is always 375GB.
+	// +optional
+	Size int64 `json:"size,omitempty"`
```

Comment:
If this was a pointer, it would be easier to set a default using `k8s.io/utils/pointer` with `pointer.Int64PtrDerefOr()`.

Reply:
Ah good point, thanks. Should probably change that for root disk size in the future too...

Reply:
Unfortunately for DiskType type alias you had to do a ton of casting so I wrote custom helper for this.